cloud_functions/utils.js

```diff
@@ -19,23 +19,45 @@ const coerceToError = e => {
   }
 }
 
-// TODO: Share this code with the bard GAE app (will probably require a more sophisticated build/deploy process).
-// However, this implementation is slightly different in that it lets exceptions escape. This is
-// appropriate for how it's used in the context of this Cloud Function, but might not be appropriate
-// for use in the GAE app or may require changes to the app. Letting exceptions escape instead of
-// returning a magic null value is a simpler, better expression of the behavior. That makes it worth
-// diverging the code now. Future de-duplication efforts should consider using this simpler
-// implementation in both places.
+// TODO: Share the code below with the bard GAE app (will probably require a more sophisticated build/deploy process).
+
+class Response {
+  constructor(status, data) {
+    this.status = status
+    this.data = data
+  }
+}
+
+const promiseHandler = fn => (req, res) => {
+  const handleValue = value => {
+    if (value instanceof Response) {
+      res.status(value.status).send(value.data)
+    } else {
+      console.error(value)
+      res.status(500).send(value.toString())
+    }
+  }
+  return fn(req, res).then(handleValue, handleValue)
+}
+
+// This implementation is slightly different from the GAE app in that it lets exceptions escape.
+// This is appropriate for how it's used in the context of this Cloud Function, but might not be
+// appropriate for use in the GAE app or may require changes to the app. Letting exceptions escape
+// instead of returning a magic null value is a simpler, better expression of the behavior. That
+// makes it worth diverging the code now. Future de-duplication efforts should consider using this
+// simpler implementation in both places.
 const getSecret = async ({ project, secretName }) =>  {
   const client = new SecretManagerServiceClient()
-  const name = `projects/${project}/secrets/${secretName}a/versions/latest`
```

Comment:
Oops. I guess I forgot to fix this after running a manual test of a failure condition.

Reply:
I'm a yuge fan of integration tests to catch these sorts of things but haven't even figured out how to do them properly in Martha yet